content/courses/program-security/pda-sharing.md

```diff
@@ -189,155 +196,168 @@ pub mod pda_sharing_recommended {
-account to a `withdraw_destination` token account.
+The `withdraw_insecure` instruction handler will transfer tokens in the `vault`
+token account to a `withdraw_destination` token account.
 
 However, as written the seeds used for signing are not specific to the vault's
-withdraw destination, thus opening up the program to security exploits. Take a
+withdrawal destination, thus opening up the program to security exploits. Take a
 minute to familiarize yourself with the code before continuing on.
 
-#### 2. Test `withdraw_insecure` instruction
+### 2. Test withdraw_insecure Instruction Handler
 
-The test file includes the code to invoke the `initialize_pool` instruction and
-then mint 100 tokens to the `vault` token account. It also includes a test to
-invoke the `withdraw_insecure` using the intended `withdraw_destination`. This
-shows that the instructions can be used as intended.
+The test file includes the code to invoke the `initialize_pool` instruction
+handler and then mint 100 tokens to the `vault` token account. It also includes
+a test to invoke the `withdraw_insecure` using the intended
+`withdraw_destination`. This shows that the instruction handlers can be used as
+intended.
 
-After that, there are two more tests to show how the instructions are vulnerable
-to exploit.
+After that, there are two more tests to show how the instruction handlers are
+vulnerable to exploit.
 
-The first test invokes the `initialize_pool` instruction to create a "fake"
-`pool` account using the same `vault` token account, but a different
+The first test invokes the `initialize_pool` instruction handler to create a
+"fake" `pool` account using the same `vault` token account, but a different
 `withdraw_destination`.
 
 The second test withdraws from this pool, stealing funds from the vault.
 
 ```typescript
-it("Insecure initialize allows pool to be initialized with wrong vault", async () => {
-  await program.methods
-    .initializePool(authInsecureBump)
-    .accounts({
-      pool: poolInsecureFake.publicKey,
-      mint: mint,
-      vault: vaultInsecure.address,
-      withdrawDestination: withdrawDestinationFake,
-      payer: walletFake.publicKey,
-    })
-    .signers([walletFake, poolInsecureFake])
-    .rpc();
-
-  await new Promise(x => setTimeout(x, 1000));
-
-  await spl.mintTo(
-    connection,
-    wallet.payer,
-    mint,
-    vaultInsecure.address,
-    wallet.payer,
-    100,
-  );
-
-  const account = await spl.getAccount(connection, vaultInsecure.address);
-  expect(Number(account.amount)).to.equal(100);
+it("allows insecure initialization with incorrect vault", async () => {
+  try {
+    await program.methods
+      .initializePool(insecureAuthorityBump)
+      .accounts({
+        pool: insecurePoolFake.publicKey,
+        mint: tokenMint,
+        vault: insecureVault.address,
+        withdrawDestination: fakeWithdrawDestination,
+      })
+      .signers([insecurePoolFake])
+      .rpc();
+
+    await spl.mintTo(
```

Comment:
Just import `mintTo`, to match the other lessons.

Reply:
PS nice work getting rid of the unnecessary delay in the old code.

Reply:
Imported required functions only and updated Unbox PRs as well for the same.